### PR TITLE
Interpolation Defaults

### PR DESCRIPTION
**Pull request by darkma773r**
Changes similar to what was recently released in commons-configuration 2.8.0.
- Make default string lookups configurable via system property. 
- Remove dns, url, and script lookups from defaults.

### CODE REVIEW

File: src/main/java/org/apache/commons/text/lookup/DefaultStringLookup.java [hunk 1]
```diff
@@ -117,14 +115,6 @@ public enum DefaultStringLookup {
      */
     XML(StringLookupFactory.KEY_XML, StringLookupFactory.INSTANCE.xmlStringLookup());
 
-    /**
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/text/lookup/DefaultStringLookup.java:120]: This is handy, why remote it?

Comment (darkma773r) [reply src/main/java/org/apache/commons/text/lookup/DefaultStringLookup.java:120]: I don't believe it is appropriate to include a `main` method here. If the values are not easily accessible via the documentation, then that indicates a failure in the documentation and not a need for a one-off tool.

**Comment by garydgregory** [comment]
It's good that the changes are similar to Commons Configuration but we should preserve the examples in Javadocs and perhaps separate them in a separate section in the Javadoc with an introduction sentence saying what you need to do to enable the relevant lookups.

**Comment by darkma773r** [comment]
@garydgregory, I've added some additional documentation. Let me know what you think.

**Comment by darkma773r** [comment]
@garydgregory, any objections to merging this?

File: src/main/java/org/apache/commons/text/lookup/DefaultStringLookup.java [hunk 2]
```diff
@@ -31,100 +34,125 @@
 public enum DefaultStringLookup {
 
     /**
-     * The lookup for Base64 decoding using the key {@value StringLookupFactory#KEY_BASE64_DECODER}.
+     * The lookup for Base64 decoding using the key {@code "base64Decoder"}.
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/text/lookup/DefaultStringLookup.java:37]: Wrong fix here and in subsequent comments, instead, use the FQCN, for example, replace:
```
{@value StringLookupFactory#KEY_BASE64_DECODER}
```
with
```
{@value org.apache.commons.text.lookup.StringLookupFactory#KEY_BASE64_DECODER}
```

**Pull request merged**